src/OrchardCore.Modules/OrchardCore.Lucene.FrenchAnalyzer/Startup.cs

```diff
@@ -0,0 +1,17 @@
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Lucene.Services;
+using OrchardCore.Modules;
+
+namespace OrchardCore.Lucene.FrenchAnalyzer
+{
+    [Feature("OrchardCore.Lucene.FrenchAnalyzer")]
+    public class Startup : StartupBase
+    {
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.Configure<LuceneOptions>(o =>
```

Comment:
Good. All the existing analyzers should be registered in the Lucene module.

Reply:
Yes, still need to add that.

Reply:
@jtkech We need to be able to do here the same as with the LocalizationMiddleware. We need to have a method to do AddAnalyzer((string)analyzerName). And probably a method to get All providers without requiring to get them from the DI each time.

Reply:
Okay as you want, here the code of `AddSupportedCultures` in `RequestLocalizationOptions.cs`.

        public RequestLocalizationOptions AddSupportedCultures(params string[] cultures)
        {
            // But i think that you don't want to loose the previuos ones.
            var supportedCultures = new List<CultureInfo>();

            foreach (var culture in cultures)
            {
                supportedCultures.Add(new CultureInfo(culture));
            }

            SupportedCultures = supportedCultures;
            return this;
        }